src/engines.jl

```diff
@@ -61,14 +64,56 @@ get_test_engine()::String = TEST_ENGINE_PROVISION.provider()
 # Release test engine. Notifies the provider that this engine is no longer in use.
 release_test_engine(name::String) = TEST_ENGINE_PROVISION.releaser(name)
 
+"""
+    set_engine_name_provider(provider::Function)
```

Comment:
the global state part of this makes some conventions a little odd, but it feels like this should be `set_engine_name_provider!` since it mutates state (though then do other things, so I'm really not sure - maybe they all should have `!`?)

Reply:
I'm not sure either. I had left it without the `!` (once I learned it existed) because it seemed more like standard operation than manipulation, but it keeps looking odd. I'll change these and see if I sleep better.

Reply:
I changed these and also exclaimed at the changes to the size of the engine pool.

Reply:
it looks a bit less odd to me this way. Looks like the names are out of sync with the docstrings now

Reply:
Synced and fixed the docstrings